### PR TITLE
ISPN-8192 Deprecate CollectionFactory

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/ISPN-8192
https://issues.jboss.org/browse/ISPN-10026
https://issues.jboss.org/browse/ISPN-10028

### CODE REVIEW

**Comment by galderz** [comment]
Needs rebase

File: commons/src/main/java/org/infinispan/commons/util/concurrent/ConcurrentHashSet.java [hunk 1]
```diff
@@ -30,23 +29,23 @@
     protected final ConcurrentMap<E, Object> map;
```
Comment (wburns) [inline commons/src/main/java/org/infinispan/commons/util/concurrent/ConcurrentHashSet.java]: To be honest this class should be deprecated as well. There is https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#newKeySet-- that can be used instead.

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/BoundedConcurrentMapNearCache.java [hunk 2]
```diff
@@ -20,8 +22,8 @@ private BoundedConcurrentMapNearCache(ConcurrentMap<K, MetadataValue<V>> cache)
    }
 
    public static <K, V> NearCache<K, V> create(final NearCacheConfiguration config) {
-      return new BoundedConcurrentMapNearCache<K, V>(
-         CollectionFactory.<K, MetadataValue<V>>makeBoundedConcurrentMap(config.maxEntries()));
```
Comment (wburns) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/BoundedConcurrentMapNearCache.java]: Sadly this is probably one of the few methods I think that might be useful to keep around. I don't think everyone should have to know how to mess with Caffeine for just a simple bounded concurrent map.

Maybe just move to `Util`?

File: core/src/main/java/org/infinispan/security/impl/AuthorizationHelper.java [hunk 3]
```diff
@@ -50,7 +52,12 @@ public AuthorizationHelper(GlobalSecurityConfiguration globalConfiguration, Audi
    }
 
    public AuthorizationHelper(GlobalSecurityConfiguration security, AuditContext context, String name) {
-      this(security, context, name, security.authorization().enabled() ? CollectionFactory.makeBoundedConcurrentMap(10) : null);
```
Comment (wburns) [inline core/src/main/java/org/infinispan/security/impl/AuthorizationHelper.java]: Same here as the other comment re: bounded concurrent map.

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/BoundedConcurrentMapNearCache.java (hunk 2 above)
Comment (danberindei) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/BoundedConcurrentMapNearCache.java]: Do we need a bounded concurrent map that often @wburns?

Comment (ryanemerson) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/BoundedConcurrentMapNearCache.java]: I don't think it's worth adding a helper method for, as we only use it twice ourselves and I can't imagine many users require it. I think we should deprecate and then provide an alternative if it's ever requested.

Comment (wburns) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/BoundedConcurrentMapNearCache.java]: Okay, it seemed cleaner not having to have the client worry about caffeine though as well. But I am fine about not having it.

File: commons/src/main/java/org/infinispan/commons/util/concurrent/ConcurrentHashSet.java (hunk 1 above)
Comment (wburns) [reply commons/src/main/java/org/infinispan/commons/util/concurrent/ConcurrentHashSet.java]: To clarify I mean the entire `ConcurrentHashSet` class

Comment (ryanemerson) [reply commons/src/main/java/org/infinispan/commons/util/concurrent/ConcurrentHashSet.java]: Added as a separate commit & jira for exposure

**Comment by ryanemerson** [comment]
@wburns I noticed that we also provide `ConcurrentWeakKeyHashMap`, but it's only actually used by the `AbstractJBossMarshaller`. I haven't looked into the implementation, but what does this provide over just using a `ConcurrentHashMap`?

**Comment by ryanemerson** [comment]
@wburns I've deprecated the class and updated `AbstractJbossMarshaller` to utilise weak keyed caffeine cache.

**Comment by wburns** [comment]
There are a lot of CI failures, I am guessing they are unrelated, but I don't know for sure https://ci.infinispan.org/job/Infinispan/job/PR-6700/6/#showFailuresLink

**Comment by ryanemerson** [comment]
@wburns Those failures were seen on one of [Dan's PRs](https://github.com/infinispan/infinispan/pull/6746#issuecomment-470426003). I've triggered a new build.

**Comment by wburns** [comment]
Integrated into master, thanks @ryanemerson !

**Pull request merged**